Error: Error refreshing state: 1 error(s) occurred:TERRAFORM CRASH

### Terraform Version

C02VVAHYHTD6:stage chenshu$ terraform --version
Terraform v0.11.3
+ provider.aws v1.17.0
+ provider.template v1.0.0

Your version of Terraform is out of date! The latest version
is 0.11.7. You can update by downloading from www.terraform.io/downloads.html

### Affected Resource(s)

C02VVAHYHTD6:stage chenshu$ terraform plan -destroy -target=module.web3-3010-patch
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_iam_role.tr_web_role: Refreshing state... (ID: tr_web_role)
aws_s3_bucket.tr_chef_config: Refreshing state... (ID: tr_chef_config)
aws_alb_target_group.tr-alb-tg: Refreshing state... (ID: arn:aws:elasticloadbalancing:us-east-1:...-tg-bearhug3010-patch/7fd6feaea4015f6e)
aws_security_group.tr_elb_sg: Refreshing state... (ID: sg-4a1bfc35)
aws_iam_instance_profile.tr_web_Iprofile: Refreshing state... (ID: tr_web_Iprofile)
aws_security_group.tr_web_sg: Refreshing state... (ID: sg-371bfc48)
aws_alb_listener.http_listener: Refreshing state... (ID: arn:aws:elasticloadbalancing:us-east-1:...-alb/9cce48264ece2ce9/04f1e2e6302b1208)

Error: Error refreshing state: 1 error(s) occurred:

* module.s3.aws_s3_bucket.tr_chef_config: 1 error(s) occurred:

* module.s3.aws_s3_bucket.tr_chef_config: aws_s3_bucket.tr_chef_config: unexpected EOF


### Debug Output

panic: interface conversion: interface {} is nil, not *s3.GetBucketCorsOutput
2018-05-21T14:16:06.404+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 
2018-05-21T14:16:06.404+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: goroutine 101 [running]:
2018-05-21T14:16:06.404+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsS3BucketRead(0xc42014c070, 0x33dae40, 0xc4200b4000, 0xc42014c070, 0x0)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_s3_bucket.go:693 +0x5c57
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc42030d110, 0xc4206b25f0, 0x33dae40, 0xc4200b4000, 0xc42068f018, 0x1000f010663c201, 0x80000000000)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:354 +0x17e
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4202751f0, 0xc4206b25a0, 0xc4206b25f0, 0x655e8b0, 0x0, 0x1000)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x9a
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc4207406a0, 0xc42060edb0, 0xc42060ee80, 0x0, 0x0)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: reflect.Value.call(0xc420376ba0, 0xc420332050, 0x13, 0x382306c, 0x4, 0xc42036af20, 0x3, 0x3, 0x0, 0x0, ...)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/usr/local/go/src/reflect/value.go:434 +0x905
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: reflect.Value.Call(0xc420376ba0, 0xc420332050, 0x13, 0xc420724720, 0x3, 0x3, 0xc420439e00, 0xc420439ec0, 0xc420237cc0)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: net/rpc.(*service).call(0xc420055480, 0xc42007a050, 0xc4203a0160, 0xc4203a4680, 0xc42017e120, 0x2ef1ca0, 0xc42060edb0, 0x16, 0x2ef1ce0, 0xc42060ee80, ...)
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: created by net/rpc.(*Server).ServeCodec
2018-05-21T14:16:06.405+0800 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2018/05/21 14:16:06 [ERROR] root.s3: eval: *terraform.EvalRefresh, err: aws_s3_bucket.tr_chef_config: unexpected EOF
2018/05/21 14:16:06 [ERROR] root.s3: eval: *terraform.EvalSequence, err: aws_s3_bucket.tr_chef_config: unexpected EOF
2018/05/21 14:16:06 [TRACE] [walkRefresh] Exiting eval tree: module.s3.aws_s3_bucket.tr_chef_config
2018-05-21T14:16:06.409+0800 [DEBUG] plugin: plugin process exited: path=/Users/chenshu/Documents/GitHub/tr_service/translation-chef-data/provisioning-vpc/stage/.terraform/plugins/darwin_amd64/terraform-provider-aws_v1.17.0_x4
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.s3.output.chef_bucket_name"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.s3.output.chef_bucket_arn"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.var.chef_bucket_name"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.data.template_file.web-userdata"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "provider.template (close)"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.aws_launch_configuration.tr_lc_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.output.tr_lc_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.aws_autoscaling_group.tr_asg_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.aws_autoscaling_attachment.asg_attachment_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "module.web3-3010-patch.output.tr_asg_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "provider.aws (close)"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "output.tr_asg_stg"
2018/05/21 14:16:06 [TRACE] dag/walk: upstream errored, not walking "root"
2018/05/21 14:16:06 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-05-21T14:16:06.411+0800 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-05-21T14:16:06.414+0800 [DEBUG] plugin: plugin process exited: path=/Users/chenshu/Documents/GitHub/tr_service/translation-chef-data/provisioning-vpc/stage/.terraform/plugins/darwin_amd64/terraform-provider-template_v1.0.0_x4



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

### Panic Output
[crash.log](https://github.com/terraform-providers/terraform-provider-aws/files/2021436/crash.log)


### Expected Behavior

terraform plan can execute successful

### Actual Behavior
terraform plan return error

### Steps to Reproduce

1. `terraform plan`

--- a/aws/resource_aws_s3_bucket.go
+++ b/aws/resource_aws_s3_bucket.go
@@ -690,14 +690,18 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 			Bucket: aws.String(d.Id()),
 		})
 	})
-	cors := corsResponse.(*s3.GetBucketCorsOutput)
 	if err != nil {
 		// An S3 Bucket might not have CORS configuration set.
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != "NoSuchCORSConfiguration" {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() != "NoSuchCORSConfiguration" {
+				return err
+			}
+			log.Printf("[WARN] S3 bucket: %s, no CORS configuration could be found.", d.Id())
+		} else {
 			return err
 		}
-		log.Printf("[WARN] S3 bucket: %s, no CORS configuration could be found.", d.Id())
 	}
+	cors := corsResponse.(*s3.GetBucketCorsOutput)
 	log.Printf("[DEBUG] S3 bucket: %s, read CORS: %v", d.Id(), cors)
 	if cors.CORSRules != nil {
 		rules := make([]map[string]interface{}, 0, len(cors.CORSRules))

